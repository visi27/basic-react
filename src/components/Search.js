import React from "react";
import Button from "./Button";

const Search = ({
  searchTerm,
  onSearchChange,
  onSearchSubmit,
  onReset,
  children
}) => (
  <form onSubmit={onSearchSubmit}>
    {children}
    <input type="text" value={searchTerm} onChange={onSearchChange} />
    <Button onClick={onReset}>RESET</Button>
  </form>
);

export default Search;
