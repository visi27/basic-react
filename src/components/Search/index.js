import React from 'react';
import Index from '../Button/';

const Search = ({
  searchTerm,
  onSearchChange,
  onSearchSubmit,
  onReset,
  children
}) => (
  <form onSubmit={onSearchSubmit}>
    {children}
    <input type="text" value={searchTerm} onChange={onSearchChange} />
    <Index onClick={onReset}>RESET</Index>
  </form>
);

export default Search;
