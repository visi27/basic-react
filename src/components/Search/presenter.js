import React from 'react';
import Button from '../Button';

const Search = ({children, searchTerm, onSearchChange}) => {
  let textInput = null;
  return (
    <form>
      {children}
      <input type="text" onChange={onSearchChange} defaultValue={searchTerm}
             ref={(input) => {
               textInput = input;
             }}/>
      <Button>RESET</Button>
    </form>
  );
};

export default Search;