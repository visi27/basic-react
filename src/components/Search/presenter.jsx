import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import Button from '../Button/resetButton';

class Search extends Component {
  componentDidMount() {
    if (this.textInput) {
      this.textInput.focus();
    }
  }

  render() {
    const { children, searchTerm, onSearchChange } = this.props;

    return (
      <form onSubmit={e => e.preventDefault()}>
        {children}
        <input
          type="text"
          defaultValue={searchTerm}
          onChange={onSearchChange}
          ref={(input) => {
            this.textInput = input;
          }}
        />
        <Button>RESET</Button>
      </form>
    );
  }
}

Search.propTypes = {
  children: PropTypes.element.isRequired,
  searchTerm: PropTypes.string,
  onSearchChange: PropTypes.func.isRequired,
};

Search.defaultProps = {
  searchTerm: '',
};

export default Search;
