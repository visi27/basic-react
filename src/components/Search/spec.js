import React from 'react';
import Search from './presenter';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('Search', () => {
  let wrapper;

  const props = {
    searchTerm: 'redux',
    onSearchChange: () => {},
    onSearchSubmit: () => {},
    onReset: () => {},
    children: 'Search'
  };

  test('has a valid snapshot', () => {
    const component = shallow(<Search {...props} />);
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
