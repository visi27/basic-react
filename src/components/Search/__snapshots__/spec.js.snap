// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form>
  Search
  <input
    defaultValue="redux"
    onChange={[Function]}
    type="text"
  />
  <button
    className="button-inline"
    type="button"
  >
    RESET
  </button>
</form>
`;
